include/SFML/Window/Event.hpp

```diff
@@ -235,30 +339,31 @@ struct Event
 ///
 /// A sf::Event instance contains the type of the event
 /// (mouse moved, key pressed, window closed, ...) as well
-/// as the details about this particular event. Please note that
-/// the event parameters are defined in a union, which means that
-/// only the member matching the type of the event will be properly
-/// filled; all other members will have undefined values and must not
-/// be read if the type of the event doesn't match. For example,
-/// if you received a KeyPressed event, then you must read the
-/// event.key member, all other members such as event.mouseMove
-/// or event.text will have undefined values.
+/// as the details about this particular event. Each event
+/// corresponds to a different struct which contains the data
+/// required to process that event.
+///
+/// The way to access the current active event is via
+/// sf::Event::getIf which returns the address of the event struct
+/// if it's active or a nullptr otherwise. sf::Event::is is used
```

Comment:
> [...] returns the address of the event struct if it's active or a nullptr otherwise.
This sounds really off. How about this one?
> [...] returns the address of the event struct, if the event type matches or is related, or a null pointer otherwise.